### PR TITLE
ENH: Suppress division by zero warning in spm_hrf

### PR DESCRIPTION
**Pull request by effigies**
`spm_hrf` generates a divide-by-zero warning on its first run. Makes for ugly output with no user action indicated.

### CODE REVIEW

File: nipype/algorithms/modelgen.py [hunk 1]
```diff
@@ -85,7 +85,10 @@ def spm_hrf(RT, P=None, fMRI_T=16):
     if P is not None:
         p[0:len(P)] = P
 
+    old_settings = np.seterr(divide='ignore')  # Known division-by-zero
```
Comment (chrisfilo) [inline nipype/algorithms/modelgen.py]: could you rephrase this to:

	with np.errstate(divide='ignore'):
	
See: http://stackoverflow.com/questions/25087769/runtimewarning-divide-by-zero-error-how-to-avoid-python-numpy

Comment (effigies) [inline nipype/algorithms/modelgen.py]: Done. Was hoping there was a context manager, but didn't see it in the docs.

Comment (effigies) [inline nipype/algorithms/modelgen.py]: Okay, now done, because I can't read on the first try.

**Comment by effigies** [comment]
Warning suppression should have been around the `_spm_Gpdf` call, not definition. Rebased to master.

**Comment by chrisfilo** [comment]
please rebase/merge master - we have fixed the CircleCI tests

**Comment by effigies** [comment]
Rebased and passing.

**Comment by effigies** [comment]
No rush, but is there anything holding up this PR?
